airline-web/app/controllers/ClientActor.scala

```diff
@@ -18,10 +25,38 @@ class ClientActor(out: ActorRef, chat: ActorRef) extends Actor {
       if (text.indexOf("[LOGGED]") > -1) {
 	   chat ! ClientSentMessage(text)
 	  } else {
-	  chat ! ClientSentMessage("[" + sdf.format(Calendar.getInstance().getTime())+ "] " + text)
+		  val json_text: JsValue = Json.parse(text)
+		  val airline = AirlineSource.loadAirlineById(userId)
+		  val otext =  "[" + sdf.format(Calendar.getInstance().getTime())+ "] " + airline.get.name + ": " + json_text.\("text").as[String]
+		  val room = json_text.\("room").as[String]
+		  var o_room = -1
+		  if (room != "chatBox-1") {
+			// Get Alliance ID
+			o_room = 0; // Default
+			val airlines = Map[Int, Airline]()
+			val airline2 = airlines.getOrElseUpdate(userId, AirlineSource.loadAirlineById(userId, false).getOrElse(Airline.fromId(userId)))
+			  AllianceSource.loadAllianceMemberByAirline(airline2).foreach { member =>
```

Comment:
Why do we need to reload the airline here? what's the map `airlines` for? The airlines.getOrElseUpdate will always run the update part as the Map is empty from the previous statement?

Reply:
It was literally the only way I could figure out how to pull the alliance id if it exists .. I'm all for a better way but I wasn't gonna touch the model .. I'll let you put in an getAllianceIDbyAirlineID

Reply:
cant u just replace airline2 with airline.get?

Reply:
I'll try it again

Reply:
Okay redone